diff --git a/Makefile b/Makefile
index 3d9707f..0d755a8 100644
--- a/Makefile
+++ b/Makefile
@@ -8,9 +8,9 @@ V8_ARCH = x64
 V8_MODE = release
 
 WASM_FLAGS = -DWASM_API_DEBUG  # -DWASM_API_DEBUG_LOG
-C_FLAGS = ${WASM_FLAGS} -Wall -Werror -ggdb -O -fsanitize=address
+C_FLAGS = ${WASM_FLAGS} -Wall -Werror -ggdb -O -fPIC
 CC_FLAGS = -std=c++11 ${C_FLAGS}
-LD_FLAGS = -fsanitize-memory-track-origins -fsanitize-memory-use-after-dtor
+LD_FLAGS =
 
 C_COMP = clang
 
@@ -45,7 +45,7 @@ EXAMPLES = \
 WASM_INCLUDE = ${WASM_DIR}/include
 WASM_SRC = ${WASM_DIR}/src
 WASM_OUT = ${OUT_DIR}/${WASM_DIR}
-WASM_C_LIBS = wasm-bin wasm-c
+WASM_C_LIBS = wasm-bin wasm-c wasm-bench
 WASM_CC_LIBS = wasm-bin wasm-v8
 WASM_C_O = ${WASM_C_LIBS:%=${WASM_OUT}/%.o}
 WASM_CC_O = ${WASM_CC_LIBS:%=${WASM_OUT}/%.o}
@@ -306,3 +306,14 @@ v8-gn-args:
 .PHONY: docker
 docker:
 	docker build -t wasm:Dockerfile .
+
+
+###############################################################################
+# Benchmarking Engine
+${OUT_DIR}/libengine.so: ${WASM_OUT}/wasm-bench.o ${WASM_C_O}
+	${CC_COMP} ${CC_FLAGS} ${LD_FLAGS} -shared $< -o $@ \
+		${WASM_OUT}/wasm-bin.o ${WASM_OUT}/wasm-c.o \
+		${LD_GROUP_START} \
+		${V8_LIBS:%=${V8_OUT}/obj/libv8_%.a} \
+		${LD_GROUP_END} \
+		-ldl -pthread
